community/modules/scheduler/schedmd-slurm-gcp-v6-controller/modules/slurm_files/scripts/resume.py

```diff
@@ -99,19 +99,32 @@ def instance_properties(nodeset, model, placement_group, labels=None):
         props.scheduling.maintenanceInterval = nodeset.maintenance_interval
 
     if nodeset.dws_flex.enabled:
-        update_props_dws(props,nodeset.dws_flex)
+        update_props_dws(props, nodeset.dws_flex, job_id)
 
     # Override with properties explicit specified in the nodeset
     props.update(nodeset.get("instance_properties") or {})
     
     return props
 
-def update_props_dws(props:dict,dws_flex:dict) -> None:
+def update_props_dws(props:object, dws_flex:object, job_id: int) -> None:
     props.scheduling.onHostMaintenance = "TERMINATE"
     props.scheduling.instanceTerminationAction = "DELETE"
-    props.scheduling.maxRunDuration['seconds'] = dws_flex.max_run_duration
     props.reservationAffinity['consumeReservationType'] = "NO_RESERVATION"
 
+    if dws_flex.use_job_duration:
+        current_job = lookup().job(job_id)
+        if not current_job or not current_job.duration:
+            props.scheduling.maxRunDuration['seconds'] = dws_flex.max_run_duration
+            return
+
+        current_job_duration = int(current_job.duration.total_seconds())
+        if 30<=current_job_duration<=1209600:
+            props.scheduling.maxRunDuration['seconds'] = current_job_duration
+            return
+
+        log.info("Job TimeLimit cannot be less than 30 seconds or exceed 2 weeks")
```

Comment:
I would encourage to not fallback to  `dws_flex.max_run_duration`. Lets sync offline.

Reply:
synced offline, will leave fall back as-is.